src/System.Private.CoreLib/shared/System/Globalization/DateTimeParse.cs

```diff
@@ -4621,6 +4636,206 @@ internal static bool TryParseQuoteString(ReadOnlySpan<char> format, int pos, Str
             return true;
         }
 
+        private static bool ParseFormatR(ReadOnlySpan<char> source, ref ParsingInfo parseInfo, ref DateTimeResult result)
+        {
+            // Example:
+            // Tue, 03 Jan 2017 08:08:05 GMT
+
+            // The format is exactly 29 characters.
+            if ((uint)source.Length != 29)
+            {
+                result.SetBadDateTimeFailure();
+                return false;
+            }
+
+            // Parse the three-letter day of week.  Any casing is valid.
+            DayOfWeek dayOfWeek;
+            {
+                uint dow0 = source[0], dow1 = source[1], dow2 = source[2], comma = source[3];
+
+                if ((dow0 | dow1 | dow2 | comma) > 0x7F)
+                {
+                    result.SetBadDateTimeFailure();
+                    return false;
+                }
+
+                uint dowString = (dow0 << 24) | (dow1 << 16) | (dow2 << 8) | comma | 0x20202000;
```

Comment:
@ahsonkhan, Utf8Parser with "r" expects the specific casing of "Sun", "Mon", etc... it doesn't allow any other casing.  Is that by design?  It differs from DateTime.ParseExact in that regard, which allows any casing for the day of week and month.